### DISCUSSION TITLE
SPM fetched binaries are quite big

### DISCUSSION THREAD

**Discussion by stherold** [braintree/braintree_ios#1106] [Q&A]
Hey hey 👋 

we recently switched from installing Braintree from Carthage to installing via SPM. We recognized the cache size increased from `1,9GB` to `2.4GB`. Fetching the binaries can take quite long. Is there any reason for that? Is it possible to reduce what is fetched by SPM?

For Carthage these are the things that have been build:
```
*** Building scheme "BraintreeAmericanExpress" in Braintree.xcworkspace
*** Building scheme "BraintreeApplePay" in Braintree.xcworkspace
*** Building scheme "BraintreeCard" in Braintree.xcworkspace
*** Building scheme "BraintreeCore" in Braintree.xcworkspace
*** Building scheme "BraintreeDataCollector" in Braintree.xcworkspace
*** Building scheme "BraintreePaymentFlow" in Braintree.xcworkspace
*** Building scheme "BraintreePayPal" in Braintree.xcworkspace
*** Building scheme "BraintreePayPalNativeCheckout" in Braintree.xcworkspace
*** Building scheme "BraintreeSEPADirectDebit" in Braintree.xcworkspace
*** Building scheme "BraintreeUnionPay" in Braintree.xcworkspace
*** Building scheme "BraintreeVenmo" in Braintree.xcworkspace
*** Building scheme "PayPalDataCollector" in Braintree.xcworkspace
*** Building scheme "BraintreeThreeDSecure" in Braintree.xcworkspace
*** Building scheme "BraintreeTestShared" in Braintree.xcworkspace
```

**Comment by scannillo** [comment]
👋 Hi @stherold - thanks for bringing this to our attention. Can you provide more detail on how you're measuring cache size for the Carthage vs SPM integrations?

**Reply by stherold** [reply]
We use Bitrise workflows cache pull/push to manage our dependency cache. For Carthage and SPM we use the following source-/lockfile mapping:
```
<project_root>/Carthage -> <project_root>/Cartfile.resolved
<project_root>/SwiftPackageManager -> <project_root>/Package.resolved
```

While `<project_root>/SwiftPackageManager` is the SPM build folder and contains the following files:

```
Tuist/Dependencies/SwiftPackageManager/.build
├── artifacts
├── checkouts
├── repositories
└── workspace-state.json
```

Then Bitrise shows the total size of the cache which was much smaller before we moved to install Braintree via SPM. 

But independent of Bitrise you can just measure the size of the Carthage/SPM folders in these 2 cases:
- install BT via Carthage
- install BT via SPM

That should be simple. By default the SPM build folder is somewhere in the derived data folder.
